pkg/clients/release/releases.go

```diff
@@ -124,3 +130,50 @@ func (r *ReleaseController) GetPipelineRunInNamespace(namespace, releaseName, re
 
 	return nil, fmt.Errorf("couldn't find PipelineRun in managed namespace '%s' for a release '%s' in '%s' namespace because of err:'%w'", namespace, releaseName, releaseNamespace, err)
 }
+
+// WaitForReleasePipelineToGetStarted wait for given release pipeline to get started.
+// In case of failure, this function retries till it gets timed out.
+func (r *ReleaseController) WaitForReleasePipelineToGetStarted(release *releaseApi.Release, managedNamespace string) (*pipeline.PipelineRun, error) {
+	var releasePipelinerun *pipeline.PipelineRun
+
+	err := wait.PollUntilContextTimeout(context.Background(), time.Second*2, time.Minute*5, true, func(ctx context.Context) (done bool, err error) {
+		releasePipelinerun, err = r.GetPipelineRunInNamespace(managedNamespace, release.GetName(), release.GetNamespace())
+		if err != nil {
+			GinkgoWriter.Println("PipelineRun has not been created yet for release %s/%s", release.GetNamespace(), release.GetName())
+			return false, nil
+		}
+		if !releasePipelinerun.HasStarted() {
+			GinkgoWriter.Println("pipelinerun %s/%s hasn't started yet", releasePipelinerun.GetNamespace(), releasePipelinerun.GetName())
+			return false, nil
+		}
+		return true, nil
+	})
+
+	return releasePipelinerun, err
+}
+
+// WaitForReleasePipelineToBeFinished wait for given release pipeline to finish.
```

Comment:
Can you talk about it printing the logs?

Reply:
Updated! Thanks